staging/src/k8s.io/client-go/tools/pager/pager.go

```diff
@@ -123,7 +123,7 @@ func (p *ListPager) List(ctx context.Context, options metav1.ListOptions) (runti
 			list.ResourceVersion = m.GetResourceVersion()
 			list.SelfLink = m.GetSelfLink()
 		}
-		if err := meta.EachListItem(obj, func(obj runtime.Object) error {
+		if err := meta.EachListItemWithAlloc(obj, func(obj runtime.Object) error {
```

Comment:
The changes to the pager methods increase the alloc cost by the number of items in the list. That happens to be balanced out by the reflect fix in reflector.go, but other callers may not be so lucky. We shouldn't change the behavior and performance of these methods, but should provide ListPager#ListWithAlloc and ListPager#EachListItemWithAlloc methods

I pushed a branch to https://github.com/liggitt/kubernetes/commits/extract_listobject_squash that does this if you want to use it

I also added documentation to all the regular and _WithAlloc methods indicating when they should be used and when callers should consider using the other one